save-payment-method/server/server.js

```diff
@@ -0,0 +1,181 @@
+import express from "express";
+import fetch from "node-fetch";
+import "dotenv/config";
+import path from "path";
+
+const { PAYPAL_CLIENT_ID, PAYPAL_CLIENT_SECRET, PORT = 8888 } = process.env;
+const base = "https://api-m.sandbox.paypal.com";
+const app = express();
+
+// host static files
+app.use(express.static("client"));
+
+// parse post params sent in body in json format
+app.use(express.json());
+
+/**
+ * Generate an OAuth 2.0 access token for authenticating with PayPal REST APIs.
+ * @see https://developer.paypal.com/api/rest/authentication/
+ */
+const authenticate = async (targetCustomerId) => {
```

Comment:
One idea is to have this function accept additional post body params in a generic way. It can have default params and allow them to be overridden or for new ones to be passed in.
```suggestion
const authenticate = async (bodyParams) => {
  const params = {
    grant_type: "client_credentials",
    response_type: "id_token",
    ...bodyParams
  };
  
  // pass the url encoded value as the body of the post call
  const urlEncodedParams = new URLSearchParams(params).toString();
```

Caller code:
```js
// normal auth for access token that uses defaults
await authenticate();

// auth for vaulting with customer id
authenticate({ target_customer_id: req.query.user });
```